lisa/analysis/frequency.py

```diff
@@ -308,7 +308,11 @@ def plot_cpu_frequencies(self, cpu, filepath=None, axis=None):
         if local_fig:
             fig, axis = self.setup_plot()
 
-        frequencies = self.trace.plat_info['freqs'][cpu]
+        try:
+            frequencies = self.trace.plat_info['freqs'][cpu]
```

Comment:
Could we split indexing on `['freqs']` (inside the `try`) and `[cpu]` (outside or in `else` clause) ?
This way we would still fail if a platform info is provided with incomplete frequencies, which would probably be the sign of a user error (if somebody wants auto-detection, then one can omit the `freqs` keys altogether).

Reply:
Would this work for you then?

```python
        if "freqs" in self.trace.plat_info:
            frequencies = self.trace.plat_info['freqs'][cpu]
        else:
            self.get_logger().info("Estimating CPU frequencies from trace")
            frequencies = df.frequency.unique()
```

Reply:
yes, that's equivalent (less efficient but nobody cares in that case)